### ISSUE TITLE
Conda environment is empty.

### ISSUE THREAD

**Issue by Hebleh** [oobabooga/one-click-installers#87]
in an older version it worked fine but then I tried reinstalling it and ever since then it stopped working giving me the Conda environment is empty, tried manually populating it, do a full install, even going back to the older version which was working but all didnt work.

**Comment by latent-variable** [comment]
I getting this error on Linux 
 ==>>>>>>>>>>>>>>>>>>>>>> ERROR REPORT <<<<<<<<<<<<<<<<<<<<<<


    Traceback (most recent call last):
      File "/home/onil/Documents/oobabooga_linux/installer_files/conda/lib/python3.10/site-packages/conda/exceptions.py", line 1132, in __call__
        return func(*args, **kwargs)
      File "/home/onil/Documents/oobabooga_linux/installer_files/conda/lib/python3.10/site-packages/conda/cli/main.py", line 69, in main_subshell
        exit_code = do_call(args, p)
      File "/home/onil/Documents/oobabooga_linux/installer_files/conda/lib/python3.10/site-packages/conda/cli/conda_argparse.py", line 120, in do_call
        module = import_module(relative_mod, __name__.rsplit('.', 1)[0])
      File "/home/onil/Documents/oobabooga_linux/installer_files/conda/lib/python3.10/importlib/__init__.py", line 126, in import_module
        return _bootstrap._gcd_import(name[level:], package, level)
      File "<frozen importlib._bootstrap>", line 1050, in _gcd_import
      File "<frozen importlib._bootstrap>", line 1027, in _find_and_load
      File "<frozen importlib._bootstrap>", line 1006, in _find_and_load_unlocked
      File "<frozen importlib._bootstrap>", line 688, in _load_unlocked
      File "<frozen importlib._bootstrap_external>", line 883, in exec_module
      File "<frozen importlib._bootstrap>", line 241, in _call_with_frames_removed
      File "/home/onil/Documents/oobabooga_linux/installer_files/conda/lib/python3.10/site-packages/conda/cli/main_create.py", line 14, in <module>
        from ..notices import notices
      File "/home/onil/Documents/oobabooga_linux/installer_files/conda/lib/python3.10/site-packages/conda/notices/__init__.py", line 3, in <module>
        from .core import notices  # noqa: F401
      File "/home/onil/Documents/oobabooga_linux/installer_files/conda/lib/python3.10/site-packages/conda/notices/core.py", line 16, in <module>
        from . import fetch
      File "/home/onil/Documents/oobabooga_linux/installer_files/conda/lib/python3.10/site-packages/conda/notices/fetch.py", line 8, in <module>
        import requests
      File "/home/onil/.local/lib/python3.10/site-packages/requests/__init__.py", line 44, in <module>
        import chardet
    ModuleNotFoundError: No module named 'chardet'


Upload did not complete.
Conda environment is empty.

**Comment by jllllll** [comment]
Does it work if you change the Miniconda version in the start script to `23.1.0-1`?

**Comment by Hebleh** [comment]
> Does it work if you change the Miniconda version in the start script to `23.1.0-1`?
Isn't it by default 23.1.0-1?

**Comment by jllllll** [comment]
The installer has been updated to use `23.3.1`

Try using the latest installer. It includes fixes for several other issues that are necessary for the latest version of the webui to function as intended. I don't know if it will fix your issue of the environment not getting created properly, but it is worth a try.

**Comment by Hebleh** [comment]
Nope, no change :(

**Comment by jllllll** [comment]
How long is the file path to the installer? A path that is too long can cause this issue on Windows.

**Comment by Hebleh** [comment]
I have tried on directly on my C drive my D drive also, but now im doing it on my downloads folder because Im getting lazy lol, but will give it a shot

**Comment by Hebleh** [comment]
installed on C:, no change

**Comment by jllllll** [comment]
Does running it as admin change anything? It could be overly-restrictive file permissions.

**Comment by Hebleh** [comment]
> Does running it as admin change anything? It could be overly-restrictive file permissions.

Nope, no change

**Comment by jllllll** [comment]
> Nope, no change

Does your user account name contain anything other than letters and numbers? Someone else recently had issues running the installer due to this.

**Comment by Hebleh** [comment]
nope just letters

> > Nope, no change
> 
> Does your user account name contain anything other than letters and numbers? Is it named "admin"? Someone else recently had issues running the installer due to this.

**Comment by Hebleh** [comment]
I got it to work on a different device, so I am suspecting that the deletion of the previous version broke something in my pc OS. I going to give it a hail mary and factory reset LOL wish me luck

**Comment by latent-variable** [comment]
> I getting this error on Linux ==>>>>>>>>>>>>>>>>>>>>>> ERROR REPORT <<<<<<<<<<<<<<<<<<<<<<
> 
> ```
> Traceback (most recent call last):
>   File "/home/onil/Documents/oobabooga_linux/installer_files/conda/lib/python3.10/site-packages/conda/exceptions.py", line 1132, in __call__
>     return func(*args, **kwargs)
>   File "/home/onil/Documents/oobabooga_linux/installer_files/conda/lib/python3.10/site-packages/conda/cli/main.py", line 69, in main_subshell
>     exit_code = do_call(args, p)
>   File "/home/onil/Documents/oobabooga_linux/installer_files/conda/lib/python3.10/site-packages/conda/cli/conda_argparse.py", line 120, in do_call
>     module = import_module(relative_mod, __name__.rsplit('.', 1)[0])
>   File "/home/onil/Documents/oobabooga_linux/installer_files/conda/lib/python3.10/importlib/__init__.py", line 126, in import_module
>     return _bootstrap._gcd_import(name[level:], package, level)
>   File "<frozen importlib._bootstrap>", line 1050, in _gcd_import
>   File "<frozen importlib._bootstrap>", line 1027, in _find_and_load
>   File "<frozen importlib._bootstrap>", line 1006, in _find_and_load_unlocked
>   File "<frozen importlib._bootstrap>", line 688, in _load_unlocked
>   File "<frozen importlib._bootstrap_external>", line 883, in exec_module
>   File "<frozen importlib._bootstrap>", line 241, in _call_with_frames_removed
>   File "/home/onil/Documents/oobabooga_linux/installer_files/conda/lib/python3.10/site-packages/conda/cli/main_create.py", line 14, in <module>
>     from ..notices import notices
>   File "/home/onil/Documents/oobabooga_linux/installer_files/conda/lib/python3.10/site-packages/conda/notices/__init__.py", line 3, in <module>
>     from .core import notices  # noqa: F401
>   File "/home/onil/Documents/oobabooga_linux/installer_files/conda/lib/python3.10/site-packages/conda/notices/core.py", line 16, in <module>
>     from . import fetch
>   File "/home/onil/Documents/oobabooga_linux/installer_files/conda/lib/python3.10/site-packages/conda/notices/fetch.py", line 8, in <module>
>     import requests
>   File "/home/onil/.local/lib/python3.10/site-packages/requests/__init__.py", line 44, in <module>
>     import chardet
> ModuleNotFoundError: No module named 'chardet'
> ```
> 
> Upload did not complete. Conda environment is empty.

I resolved this issue by deleting the python version located /home/onil/.local/lib/python3.10. I am not sure by it would try to use the "request" package located there instead of the one inside of the _installer_files_

**Comment by Hebleh** [comment]
factory resetting seems to have fixed it, once I get it working with the bot Im using ooba for ill update and close the thread out thanks alot @jllllll for helping!

**Comment by Hebleh** [comment]
all good thanks!

**Comment by Andie-Squirrel** [comment]
Hi, I'm having this issue on Windows 10 (RTX 3080Ti0 and it seemed to only happen once another installation was removed for some other arbitrary issue.

After installing a new version, the webui and generation worked perfectly. The next time I ran it, it gave me a `Conda environment is empty.` error.
This had also happened when my Antivirus flagged some process in Oobabooga as malicious, and I had to add it to the exclusions list as it gave me the same `Conda environment is empty` error. Now that I have it in exclusions, I wasn't expecting a brand new installation to be giving me this error.

I cannot afford to factory reset this machine, so any help will be greatly appreciated! I don't have enough disk space and time to infinitely keep installing new versions of Oobabooga on my machine. Thanks!

**Comment by jllllll** [comment]
@Andie-Squirrel That message occurs if `\installer_files\env\python.exe` is missing. In the past, this was due to the environment creation silently failing, hence the message. More recently, some antivirus software has gotten more sensitive and overprotective. I assume this is due to the incorporation of AI detection systems. Essentially, these systems will flag and delete any files that aren't seen on a majority of user's systems. Sometimes even files that are in different locations can trigger it.

In this case, as in many cases that I've seen, your antivirus has deleted Python from the environment. If you add it as an exclusion, keep in mind that the exclusion only applies to that exact file path and installing it elsewhere will trigger the deletion again. If you antivirus allows it, you should exclude a higher-level folder that the file is contained in instead of just the one file.

If this doesn't work, then you will have to disable that antivirus as there is nothing we can do about it on our end.

**Comment by Andie-Squirrel** [comment]
@jllllll The legendary jllllll! I was wondering if it might have deleted python.exe. I suspected the AV would do so, and I had added a higher-level folder that contained all my Oobabooga installations, yet even after installing another version it was still deleted. pythonw.exe still exists, but I don't know how my AV gets past exclusions and manages to delete the python.exe file itself just after running the webui for the first time.

Looks like I'm going to have to do disable it as you said, or exclude the python.exe file specifically if the AV will listen to me for once, and then reinstall.

Thank you very much for the advice on the in-place upgrade! I wasn't aware of this before and I'm glad that it was able to help you with your PC issues. Also, just wanted to take a quick moment to thank you for your one-click installer some time ago. I wasn't able to get the normal version of Oobabooga running at the time but your version has helped a ton in the past, so thank you very much again!
